### PR TITLE
Fix IdentitySerde

### PR DESCRIPTION
**Pull request by srkukarni**
### Motivation

We need to make sure that IdentitySerde(which is the default serde for python) outputs bytes on serialization. 
### Modifications

Describe the modifications you've done.

### Result

After your change, what will change.

### CODE REVIEW

File: pulsar-client-cpp/python/pulsar/functions/serde.py [hunk 1]
```diff
@@ -68,9 +68,9 @@ def deserialize(self, input_bytes):
       return pickle.loads(input_bytes)
 
 class IdentitySerDe(SerDe):
-  """Pickle based serializer"""
+  """utf-8 based identity serializer"""
```
Comment (sijie) [inline pulsar-client-cpp/python/pulsar/functions/serde.py]: I don't fully understand here. "identity" typically means not changing things, but the code here is converting between bytes and string, which doesn't seem to be correct to me.

Comment (srkukarni) [reply pulsar-client-cpp/python/pulsar/functions/serde.py]: I think IdentitySerde is a bad name(probably should have been called simple serde).
I didn't rename it to preserve backwards compatibility(not sure whether its worth preserving)

File: pulsar-client-cpp/python/pulsar/functions/serde.py [hunk 2]
```diff
@@ -68,9 +68,9 @@ def deserialize(self, input_bytes):
...
   def serialize(self, input):
-    return input
+    return bytes(str(input).encode('utf-8'))
```
Comment (jerrypeng) [inline pulsar-client-cpp/python/pulsar/functions/serde.py]: how can we assume that the string representation of the object is meaningful?  The identity serde is also used as the default serde in python which is going to cause errors if users inputs/outputs anything other than Strings

Comment (jerrypeng) [reply pulsar-client-cpp/python/pulsar/functions/serde.py]: by default the string representation of an object is:

>>> str(foo)
'<__main__.Foo instance at 0x10631d320>'

File: pulsar-client-cpp/python/pulsar/functions/serde.py (hunk 1 above)
Comment (srkukarni) [reply pulsar-client-cpp/python/pulsar/functions/serde.py]: Lets step back to what we are trying to accomplish. The whole idea is to have a default serializer so users dont have to specify their own serializer. Should we rename IdentitySerde to StringSerDe and make that the default?

**Comment by srkukarni** [comment]
@sijie @jerrypeng I've changed IdentitySerDe to StringSerDe to better reflect what it's role is. Please take a look again.
Thanks!

**Comment by jerrypeng** [comment]
@srkukarni why is the StringSerDe the default though?  As I mentioned in my previous comment,  if the default is StringSerDe, it will break if the user returns a class.  Shouldn't the default be the pickle serde that transforms all objects into bytes?

**Comment by srkukarni** [comment]
@jerrypeng Pickle format is very specific to Python. My main concern about making it default is that out of the box java and python functions will become illegible to each other.

**Comment by jerrypeng** [comment]
@srkukarni then we should serialize python objects to json like the JSONSchema in java. Something like:

https://stackoverflow.com/questions/10252010/serializing-class-instance-to-json

**Comment by srkukarni** [comment]
@jerrypeng since python does not support schema yet, the above won't be possible because during deserialization, we won't know what to transform into.

**Comment by jerrypeng** [comment]
@srkukarni gotcha we still need one serializer that can take care of primitive types as well as classes

**Comment by jerrypeng** [comment]
Using pickle might not be a bad thing for this before schema is implemented for python.  Its analogous to the java default serializer. Without schema its pointless to go between languages anyways, because you won't know the data type.

**Comment by srkukarni** [comment]
@jerrypeng I've added selective checks for IdentitySerDe. This kind of mimics the Java world as well as the very purpose of a default Serde. Please take a look again.
Thanks!

**Pull request merged**